File: Extensions.Tests/AsCharsTests.fs
namespace Tests.Extensions

open System
open System.Text
open Stringier
open Microsoft.VisualStudio.TestTools.UnitTesting

[<TestClass>]
type AsCharsTests() =
    [<DataTestMethod>]
    [<DataRow([|'A'|], 0x41)>]
    [<DataRow([|'Þ'|], 0xDE)>]
    [<DataRow([|'ö'|], 0xF6)>]
    [<DataRow([|'Ξ'|], 0x39E)>]
    [<DataRow([|'℥'|], 0x2125)>]
    [<DataRow([|'⎃'|], 0x2383)>]
    [<DataRow([|'\uD834'; '\uDD1E'|], 0x1D11E)>] // 𝄞 which can't be represented with a single char
    member _.``rune AsChars`` (exp:char[], value:Int32) =
        CollectionAssert.AreEqual(exp, Rune(value).AsChars())

File: Tests/AsCharsTests.cs
using System;
using System.Text;
using Stringier;
using Defender;
using Xunit;

namespace Tests {
	public class AsCharsTests : Trial {
		[Theory]
		[InlineData(0x41, new[] { 'A' })]
		[InlineData(0xDE, new[] { 'Þ' })]
		[InlineData(0xF6, new[] { 'ö' })]
		[InlineData(0x039E, new[] { 'Ξ' })]
		[InlineData(0x2125, new[] { '℥' })]
		[InlineData(0x2383, new[] { '⎃' })]
		[InlineData(0x1D11E, new[] { '\uD834', '\uDD1E' })] // 𝄞 which can't be represented with a single char
		public void Rune_AsChars(Int32 value, Char[] expected) => Claim.That(new Rune(value).AsChars()).SequenceEquals(expected);
	}
}
